hazelcast/src/main/java/com/hazelcast/map/impl/operation/steps/PutOpSteps.java

```diff
@@ -87,45 +88,47 @@ public void runStep(State state) {
             MapContainer mapContainer = recordStore.getMapContainer();
             MapServiceContext mapServiceContext = mapContainer.getMapServiceContext();
 
-            if (staticParams.isPutIfAbsent()) {
-                Record record = recordStore.getRecord(state.getKey());
-                if (record == null && state.getOldValue() != null) {
-                    record = ((DefaultRecordStore) recordStore).onLoadRecord(state.getKey(),
-                            state.getOldValue(), false, state.getCallerAddress());
-                }
-
-                if (record != null) {
+            if (!staticParams.isPutVanilla()) {
+                if (staticParams.isPutIfAbsent()) {
+                    Record record = recordStore.getRecord(state.getKey());
+                    if (record == null && state.getOldValue() != null) {
+                        record = ((DefaultRecordStore) recordStore).onLoadRecord(state.getKey(),
+                                state.getOldValue(), false, state.getCallerAddress());
+                    }
+
+                    if (record != null) {
+                        state.setOldValue(record.getValue());
+                        state.setStopExecution(true);
```

Comment:
Is this just refactoring, or was this change needed to fix something?

Reply:
no fix, it is about aligning it with no-offload case.